src/stan-users-guide/simple_linear_regression/regression_efficient.stan

```diff
@@ -0,0 +1,19 @@
+data {
+  int<lower=0> N;
+  vector[N] x;
+  vector[N] y;
+}
+
+parameters {
+  real alpha_hat;
```

Comment:
Parameters should not be suffixed `_hat`.  The notation alpha-hat is used for a point estimate of the parameter `alpha`, so the parameter is just `alpha`. (This gets subtle when you define things like y-hat in the model, but my recommendation is to never call anything hat in the model.)

Reply:
I didn't realize _hat designated a point estimate. I wanted to distinguish from the generating programs, can I call them alpha_truth or something else? I think it is confusing to have the same variable have very different meanings.

Reply:
Overloading is the way statisticians other than probability theorists roll w.r.t. naming.  It's very confusing for beginners.  I used to complain about this to no end.

First, the variables in the model should absolutely just be called `alpha`, `beta`, etc.  Anyting else is going to be too confusing to users.  Then their posterior means would be called alpha-hat (the posterior mean is the standard Bayesian estimator of a parameter as it minimizes expected square error).

I'd also just call the simulated value `alpha` in the R program.  The acceptable alternative would be `alpha_sim` but it starts getting clunky fast when you start decorating uses of `alpha` (hence the overloaded notation in the first place).

Reply:
How about alpha_s on the data generation side? I suffered at the hands of this particular convention and I'd like to not perpetuate it.